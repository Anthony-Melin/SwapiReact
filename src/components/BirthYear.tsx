import { PropsWithChildren } from 'react'

type BirthYearProps = PropsWithChildren<{ value: string }>

const BirthYear = ({ value, ...restProps }: BirthYearProps) => {
    return (
        <p data-testid="birthyear" {...restProps}>
            Birth year: {value}
        </p>
    )
}

export default BirthYear
