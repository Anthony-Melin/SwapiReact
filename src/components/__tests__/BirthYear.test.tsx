import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import BirthYear from '../BirthYear.tsx'

describe('BirthYear', () => {
    beforeEach(() => {
        render(<BirthYear value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('birthyear')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('birthyear')).toHaveTextContent(
            'Birth year: VALUE'
        )
    })
})
