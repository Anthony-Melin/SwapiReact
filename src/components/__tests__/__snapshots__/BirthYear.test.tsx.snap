// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BirthYear > should match snapshot 1`] = `
<p
  data-testid="birthyear"
>
  Birth year: 
  VALUE
</p>
`;
